### PR TITLE
Bug 2001413: add/delete nic and disk to template

### PR DESCRIPTION
**Pull request by gouyang**
note for review:
1. rename tests/disks to tests/disk, tests/vm-templates to tests/template

### CODE REVIEW

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/create-template-from-wizard.spec.ts [hunk 1]
```diff
@@ -59,7 +59,7 @@ const pxeTemplate: VirtualMachineData = {
   networkInterfaces: [nic0],
 };
 
-describe('Test VM creation', () => {
+describe('Test Template creation', () => {
```
Comment (pcbailey) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/create-template-from-wizard.spec.ts:62]: nitpick: `Template` shouldn't be capitalized here.

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  name: 'nic-1',
+  nad: 'bridge-network',
+};
+
+const disk1: Disk = {
+  name: 'disk-1',
+  size: '1',
+};
+
+describe('Test Template resource', () => {
```
Comment (pcbailey) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts:18]: nitpick: `Template` shouldn't be capitalized here.

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    deleteRow(nic1.name);
+  });
+
+  it('ID(CNV-1849) Add/Delete a disk to a VM template', () => {
+    tab.navigateToDisk();
+    addDisk(disk1);
+    deleteRow(disk1.name);
+  });
+
+  it('ID(CNV-7233) Delete user created template', () => {
```
Comment (pcbailey) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts:57]: nitpick: 'user created' => 'user-created'

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+    cy.deleteResource({
+      kind: 'Template',
+      metadata: {
+        name: TEMPLATE.DEFAULT.name,
+        namespace: testName,
+      },
+    });
+  });
+
+  it('ID(CNV-1850) Add/Delete a NIC to a VM template', () => {
```
Comment (pcbailey) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts:45]: nitpick: 'to' => 'to/from'

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+    tab.navigateToNetwork();
+    addNIC(nic1);
+    deleteRow(nic1.name);
+  });
+
+  it('ID(CNV-1849) Add/Delete a disk to a VM template', () => {
```
Comment (pcbailey) [inline frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts:51]: nitpick: 'to' => 'to/from'

**Review by pcbailey — CHANGES_REQUESTED** [review]
Just a few nitpicks. I would have went ahead and approved it and say we could fix them in a followup, but it needs a rebase to fix the `errorOccurred` change from https://github.com/openshift/console/pull/10017 as well.

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/create-template-from-wizard.spec.ts (hunk 1 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/create-template-from-wizard.spec.ts:62]: NP

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts (hunk 2 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts:18]: NP

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts (hunk 3 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts:57]: NP

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts (hunk 4 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts:45]: NP

File: frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts (hunk 5 above)
Comment (gouyang) [reply frontend/packages/kubevirt-plugin/integration-tests-cypress/tests/template/template-resource.spec.ts:51]: NP

**Comment by gouyang** [comment]
@yaacov @pcbailey PTAL, thanks!
Note: I reverted the fix of flaky for tests/smoke/actions, added a clear wait `cy.wait(5000);` to `vm.start()` and `vm.restart()`.

**Pull request merged**